lib/util/code-level-metrics.js

```diff
@@ -19,33 +20,37 @@ function setFunctionName(name) {
 }
 
 /**
- * Helper to retrieve Code Level Metrics(CLM)
- * properties from a function reference.
+ * Helper used to assign Code Level Metrics(CLM)
+ * to an active segment.
+ *
+ * spec states if function or filepath are > 255, do not assign
+ * CLM attrs
  *
  * @param {Function} fn function reference
- * @returns {object} CLM properties
+ * @param {TraceSegment} segment active segment to attach code.* attrs
  */
-module.exports = function getCLMMeta(fn) {
+module.exports = function addCLMAttributes(fn, segment) {
+  if (!fn[symbols.clm]) {
+    return
+  }
+
   try {
     const { funcInfo } = require('@contrast/fn-inspect')
-    const { lineNumber, method, file: filePath } = funcInfo(fn)
+    const { lineNumber, method, file: filePath, column } = funcInfo(fn)
     const fnName = setFunctionName(method)
 
     if (isValidLength(fnName, 255) && isValidLength(filePath, 255)) {
-      return {
-        'code.filepath': filePath,
-        'code.function': fnName,
-        'code.lineno': lineNumber + 1 // line numbers start at 0 in v8 so we have to add 1 to reflect js code
-      }
+      segment.addAttribute('code.filepath', filePath)
+      segment.addAttribute('code.function', fnName)
+      segment.addAttribute('code.lineno', lineNumber + 1) // line numbers start at 0 in v8 so we have to add 1 to reflect js code
+      segment.addAttribute('code.column', column)
```

Comment:
Is column also 0 based indexing? I didn't see anything about it on https://v8.dev/docs/stack-trace-api

Reply:
Also: did we land on `column` vs `columnno`? I know there was discussion about keeping it consistent with `lineno`.

Reply:
we don't have to add 1 to it. `columnno` reads weird to me which is why i changed to column, we are driving this